src/c-unchop.c

```diff
@@ -300,6 +446,318 @@ r_obj* list_unchop_fallback(r_obj* ptype,
 
   out = KEEP(vec_slice_fallback(out, locations));
 
-  FREE(7);
+  FREE(8);
+  return out;
+}
+
+static
+r_obj* list_unchop_c_fallback_invoke(
+  r_obj* xs,
+  r_obj* default_,
+  r_obj* name_spec,
+  struct r_lazy error_call
+) {
+  if (default_ != r_null) {
+    xs = push_default(xs, default_);
+  }
+  KEEP(xs);
+
+  r_obj* out = vec_c_fallback_invoke(xs, name_spec, error_call);
+
+  FREE(1);
+  return out;
+}
+
+// This is `vec_c_fallback()` with support for `default`.
+// Keep aligned with `vec_c_fallback()`!
+// TODO: Let `list_unchop()` own fallback helpers and
+// make `vec_c_fallback()` and friends thin wrappers
+// that set `default` to `r_null`, or merge `vec_c()`
+// and `list_unchop()` entirely!
+static
+r_obj* list_unchop_c_fallback(
```

Comment:
Really I think that most of the fallback helpers should live in `c-unchop.c` not `c.c`, because `c-unchop.c` has the more general definition with a possible `default_` argument.

We can then just make wrappers around these to create the `c.c` versions that set `default_` to `r_null`, and then we won't have this duplication. But this was noisy enough that I didn't want to do it right now.